tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/Refiner.scala

```diff
@@ -69,10 +69,12 @@ final class Refiner(semantics: Semantics, outputMode: OutputMode) {
     }
 
     val kind = {
-      if (classDef.kind == ClassKind.ModuleClass && !info.isModuleAccessed)
-        ClassKind.Class
-      else
+      if (classDef.kind.hasModuleAccessor && !info.isModuleAccessed) {
+        if (classDef.kind == ClassKind.ModuleClass) ClassKind.Class
+        else ClassKind.JSClass
+      } else {
         classDef.kind
+      }
```

Comment:
> ClassKind's companion object?

Yeah... I guess that's best. It is a step in the right direction probably. You probably want to keep it private to `optimizer` though...

Btw. Is it really a good idea to have a `hasModuleAccessor` flag? This will lead to combinations which are not allowed... (anyways, not subject to discussion right now).